### PR TITLE
managedad - Solution that creates AWS Managed Microsoft AD directory that runs Active Directory (AD)

### PR DESCRIPTION
**Pull request by tekdj7**
*Description of changes:*
This solution creates an AWS Managed Microsoft AD directory that runs Active Directory (AD).

- (Optional) Create AWS resources (IAM role & instance profile) to support seamlessly join Windows EC2 instances to your AWS Managed Microsoft AD directory.
- (Optional) Create AWS resources (IAM role, instance profile, and secret) to support seamlessly join Linux EC2 instances to your AWS Managed Microsoft AD directory.
- (Optional) Creates a Domain Members Security Group with EXAMPLE rules allowing all Private IP communications inbound.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws/solutions/ManagedAD/templates/MANAGEDAD.cfn.yaml [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+      VPCID:
+        default: VPC ID
+Parameters:
+  AWSManagedADDomainDNSName:
+    AllowedPattern: '[a-zA-Z0-9-]+..+'
+    Default: awsmad.lab
+    Description: Fully qualified domain name for the AWS Managed Microsoft AD directory, such as corp.example.com.
+    Type: String
+  AWSManagedADDomainNetBiosName:
+    AllowedPattern: '[a-zA-Z0-9\-]+'
```
Comment (jamesoff) [inline aws/solutions/ManagedAD/templates/MANAGEDAD.cfn.yaml:61]: No need to escape the last hyphen here.

File: aws/solutions/ManagedAD/templates/MANAGEDAD.cfn.yaml [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+        default: Private Subnet 2 ID
+      SecretsManagerDomainCredentialsSecretsKMSKey:
+        default: Secrets Manager KMS Key for domain credentials secret
+      SSMLogsBucketName:
+        default: Systems Manager (SSM) Logs Bucket Name
+      VPCID:
+        default: VPC ID
+Parameters:
+  AWSManagedADDomainDNSName:
+    AllowedPattern: '[a-zA-Z0-9-]+..+'
```
Comment (jamesoff) [inline aws/solutions/ManagedAD/templates/MANAGEDAD.cfn.yaml:56]: Should the first `.` be escaped?

File: aws/solutions/ManagedAD/templates/MANAGEDAD.cfn.yaml [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+    Description: Create AWS resources (IAM role & instnace profile)to support seamless domain join Windows EC2 instances
+    Type: String
+  PrivateSubnet1ID:
+    Description: ID of the private subnet 1 in Availability Zone 1 (e.g., subnet-a0246dcd)
+    Type: AWS::EC2::Subnet::Id
+  PrivateSubnet2ID:
+    Description: ID of the private subnet 2 in Availability Zone 2 (e.g., subnet-a0246dcd)
+    Type: AWS::EC2::Subnet::Id
+  SecretsManagerDomainCredentialsSecretsKMSKey:
+    AllowedPattern: '^$|^arn:(aws[a-zA-Z-]*)?:kms:[a-z0-9-]+:\d{12}:key\/[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$'
```
Comment (jamesoff) [inline aws/solutions/ManagedAD/templates/MANAGEDAD.cfn.yaml:106]: Should the first group here (the partition) be optional?

Comment (tekdj7) [reply aws/solutions/ManagedAD/templates/MANAGEDAD.cfn.yaml:106]: it should not, I will correct it.

**Pull request merged**